wiki/Rupture-Propagation.md

```diff
@@ -0,0 +1,176 @@
+
+To fairly sample rupture propagation trees, we calculate the **conditional probability** $P(T | T \text{ is a spanning tree of } G)$, which is given by the formula:
+
+$$
+P(T | T \text{ is a spanning tree of } G) = \frac{P(T)}{\sum_{T^\prime \text{ is a spanning tree of } G} P(T^\prime)}
+$$
+
+For realistic scenarios, including the Darfield earthquake, there are thousands or tens of thousands of spanning trees, but most have a trivial probability. We want to sample these spanning trees according to their conditional probability **without** generating every single one.
+
+## Handling the Initial Fault
+
+The initial fault could be known, unknown, or determined according to a prior distribution over the nodes. To account for this, we independently select one of the sources to be the initial fault after selecting a spanning tree. Selecting an initial fault and tree completely determines the **rupture propagation tree**, which is defined as a spanning tree with directed edges indicating the direction of rupture propagation. For example, in the graph $H$, if we label the nodes $A$ through $E$ and select $A$ as the initial source, we get the following rupture propagation tree:
+
+![Untitled-2024-06-24-1457(11)](images/rup_prop/initial_fault.png)
+
+## A Toy Example
+
+To get a feel for this problem, let’s create a simple example with three point sources. Assume we have three point sources $A$, $B$, and $C$, and the probability of a rupture jumping between $A$ and $B$ is $80\%$, between $B$ and $C$ is $90\%$, and between $A$ and $C$ is $30\%$. Ignoring the direction for a moment, there are three possible rupture trees $T_1$, $T_2$, and $T_3$, each with different probabilities:
+
+![Untitled-2024-06-24-1457(7)](images/rup_prop/toy_example.png)
+
+The conditional probabilities for $T_1$, $T_2$, and $T_3$ are roughly $0.18$, $0.73$, and $0.08$, respectively. If we generate 100 realisations for these three sources, we should expect $T_1$ in 18-19 cases, $T_2$ in 73–74 cases, and $T_3$ in 8–9 cases.
+
+## The Sampling Algorithm
+
+We can only efficiently sample spanning trees proportional to their **weight**, which is defined as the product of edge weights in the tree. The probability distribution we obtain is:
+
+$$
+P_w(T) = \frac{\prod_{e \in T} w(e)}{\sum_{T^\prime} \prod_{e \in T^\prime} w(e)}
+$$
+
+At first glance, $P_w(T)$ looks identical to $P(T)$, but alas, it isn’t. Instead, it over-represents unlikely rupture scenarios! In the toy example above, the probabilities for $T_1$, $T_2$, and $T_3$ become roughly $0.22$, $0.59$, and $0.20$, respectively. You can see that $T_3$ would be oversampled in 100 realisations relative to its true probability. To make this work, we construct a different graph $G^\prime$, where the edge weights are adjusted as:
```

Comment:
If I understand correctly $P_w(T)$ is different from $P(T)$ as it doesn't account for the edge weights that aren't included? I.e. the $\prod_{e \notin H} (1 - w(e))$ term. Might be worth mentioning here.

Reply:
Yeah that's the correct intuition. Formally, $P_w(H)$ is the probability that a subgraph of $G$ contains $H$. The conditional probability $P_w(T | T \text{ is a spanning tree of } G)$ is then the probability that a subgraph containing a spanning tree contains $T$. This measure will not penalise $T$ for not containing likely jumps.